runtime/libc/wasi_sdk_backing.c

```diff
@@ -318,7 +332,17 @@ wasi_errno_t wasi_snapshot_preview1_environ_sizes_get(
     wasi_size_t environc_retptr, 
     wasi_size_t environv_buf_len_retptr
 ) {
-    wasi_unsupported_syscall(__func__);
+    extern char **environ;
+
+    for (char **cursor = environ; *cursor != NULL; cursor++) {
+        runtime_environc++;
+        runtime_environv_buf_len += (strlen(*cursor) + 1);
```

Comment:
Why are you modifying globals in what should be a stateless function?

Reply:
The `WASI` syscall `environ_get` is strange because it long passes the base address of a buffer and not the length. According to the spec, the logic for this is `The sizes of the buffers should match that returned by environ_sizes_get.` That means that a WASI program must call `environ_args_get` before `environ_get`. Because of this dependency, I'm using globals to avoid having to re-execute the logic of `environ_sizes_get` in `environ_get`. If this bugs you, I could potentially create a utility function that memoizes the stat of runtime_environc and runtime_environv_buf_len into static locals.